Tasks/Common/docker-common/containerimageutils.ts

```diff
@@ -24,4 +24,41 @@ export function generateValidImageName(imageName: string): string {
     imageName = imageName.toLowerCase();
     imageName = imageName.replace(/ /g,"");
     return imageName;
-}
\ No newline at end of file
+}
+
+export function getBaseImageName(contents: string): string {
+    var lines = contents.split(/[\r?\n]/);
+    var i;
+    for (i = 0; i < lines.length; i++) {
+        var index = lines[i].toUpperCase().indexOf("FROM");
+        if (index != -1) {
+            var rest = lines[i].substring(index + 4);
+            var imageName = rest.trim();
+            return imageName;
+        }
+    }
+    
+    return null;
+}
+
+export function getResourceUrl(image: string, digest: string) {
+    var match = image.match(/^(?:([^\/]+)\/)?(?:([^\/]+)\/)?([^@:\/]+)(?:[@:](.+))?$/);
+    if (!match) {
+        return null;
+    }
+
+    var registry = match[1];
+    var namespace = match[2];
+    var repository = match[3];
+    var tag = match[4];
+  
+    if (!namespace && registry && !/[:.]/.test(registry)) {
+      namespace = registry
+      registry = null
+    }
+
+    registry = registry ? registry : 'docker.io';
+    namespace = namespace ? namespace : 'library';
+
+    return "https://" + registry + "/" + namespace + "/" + repository + "@sha256" + digest;
```

Comment:
when repository is null, this could be a problem
use format, that will take care of null values properly